server_addon/houdini/client/ayon_houdini/api/hda_utils.py

```diff
@@ -418,3 +421,74 @@ def keep_background_images_linked(node, old_name):
 
     if changes:
         set_background_images(parent, images)
+
+
+def get_products_menu_items(node):
+    """Return products menu items
+
+    It gets a list of available products of the specified product types
+      within the specified folder path with in the specified project.
+
+    Users can specify those in the HDA parameters.
+
+    Returns:
+        List[str]: Product options for Products menu.
+    """
+    project_name = node.evalParm("project_name")
+    folder_path = node.evalParm("folder_path")
+    representation_name = node.evalParm("representation_name")
+
+    folder_entity = ayon_api.get_folder_by_path(project_name,
+                                                folder_path,
+                                                fields={"id"})
+    if not folder_entity:
+        return []
+
+    products = ayon_api.get_products(
+        project_name,
+        folder_ids=[folder_entity["id"]],
+        product_types=[representation_name]
```

Comment:
I think we're better off removing this filter.

1. Technically it's incorrect, the representation name is not required to be 1:1 the product type - it may often be the case but it's not a given. You may have a `render` product type that is `exr`, you may have a `pointcache` product type with `abc` representation.
2. The filtering will likely also mean that if representation name is currently empty value on the node that it will basically return nothing here.

Reply:
oh, I get it now. thanks for the explanation.

Reply:
So, the correct filter would be: Get me all products whose representation is e.g. `usd` or `abc` in a particular folder entity ? 

I gave it a quick shot and I think such a filter won't return products based on their representations. Instead, it will provide a pair of each product and the specific version that includes that representation.

Reply:
I personally think it's weird in the the parms you have that representation comes above products - it's basically ALWAYS the other way around in all queries in the API.

As such, I also wouldn't do filtering to representation on the product.

What we _could do_ is have a "Product Type" field and add a docstring that it's solely used to filter? Yet as soon as you set a representation and it updates that it also updates the Product Type field. That way we can still have an explicit filtering field - even if it's just for UI filtering?

Reply:
That makes much sense. thank you.

Reply:
Filter is removed https://github.com/BigRoy/ayon-core/pull/9/commits/56c02d58a38afbb7c6fb263362c3554621abbffe